ray_lightning/ray_horovod.py

```diff
@@ -63,39 +67,94 @@ class HorovodRayPlugin(HorovodPlugin):
             from ray_lightning import HorovodRayPlugin
 
             ptl_model = MNISTClassifier(...)
-            plugin = HorovodRayPlugin(num_workers=2, use_gpu=True)
+            strategy = HorovodRayPlugin(num_workers=2, use_gpu=True)
 
             # Don't set ``gpus`` in ``Trainer``.
             # The actual number of GPUs is determined by ``num_workers``.
-            trainer = pl.Trainer(..., plugins=[plugin])
+            trainer = pl.Trainer(..., strategy=strategy)
             trainer.fit(ptl_model)
 
     """
+    strategy_name = "horovod_ray"
 
     def __init__(self,
                  num_workers: int,
                  num_cpus_per_worker: int = 1,
                  use_gpu: bool = False):
 
         if not HOROVOD_AVAILABLE:
-            raise RuntimeError("Please intall Horovod to use this plugin.")
+            raise RuntimeError("Please intall Horovod to use this strategy.")
         if not ray.is_initialized():
             ray.init()
-        super().__init__()
-        self.nickname = "horovod_ray"
+        ParallelStrategy.__init__(
+            self, accelerator="_gpu" if use_gpu else "cpu")
         self.num_workers = num_workers
         self.cpus_per_worker = num_cpus_per_worker
         self.use_gpu = use_gpu
         self.executor = None
+        self._exit_stack = None
+        self._local_rank = 0
+
+        self._is_remote = False
 
-    def __getstate__(self):
-        d = self.__dict__.copy()
-        del d["executor"]
-        return d
+    def _configure_launcher(self):
```

Comment:
in the pytorch lightning trainer, if there is a function `_configure_launcher` there, it will be activated from here: https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/trainer/connectors/accelerator_connector.py#L804